stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+        """
+        max_post_idx = jnp.argmax(self.Results.log_posterior_density)
+        map_points = jax.tree_map(lambda x: x[max_post_idx], self.Results.samples)
+
+        return map_points
+
+    def get_max_likelihood_parameters(self):
+        """
+        Retruns the maximum likelihood parameters
+        """
+        max_like_idx = jnp.argmax(self.Results.log_L_samples)
+        max_like_points = jax.tree_map(lambda x: x[max_like_idx], self.Results.samples)
+
+        return max_like_points
+
+    def posterior_plot(self, name: str, n=0, axis=None, save=False, filename=None):
+        """
+        Plots the posterior histogram for the given parameter
+
+        Parameters
+        ----------
+        name : str
+            Name of the parameter.
+            Should be from the names of the parameter list used or from the names of parameters
+            used in the prior_function
+
+        n : int, default 0
+            The index of the parameter to be plotted.
+            For multivariate parameters, the index of the specific parameter to be plotted.
+
+        axis : list, tuple, string, default ``None``
+            Parameter to set axis properties of ``matplotlib`` figure. For example
+            it can be a list like ``[xmin, xmax, ymin, ymax]`` or any other
+            acceptable argument for ``matplotlib.pyplot.axis()`` method.
+
+        save : bool, optional, default ``False``
+            If ``True``, save the figure with specified filename.
+
+        filename : str
+            File name and path of the image to save. Depends on the boolean ``save``.
+
+        Returns
+        -------
+        plt : ``matplotlib.pyplot`` object
+            Reference to plot, call ``show()`` to display it
+
+        """
+        nsamples = self.Results.total_num_samples
+        samples = self.Results.samples[name].reshape((nsamples, -1))[:, n]
+        plt.hist(
+            samples, bins="auto", density=True, alpha=1.0, label=name, fc="None", edgecolor="black"
+        )
+        mean1 = jnp.mean(self.Results.samples[name])
+        std1 = jnp.std(self.Results.samples[name])
+        plt.axvline(mean1, color="red", linestyle="dashed", label="mean")
+        plt.axvline(mean1 + std1, color="green", linestyle="dotted")
+        plt.axvline(mean1 - std1, linestyle="dotted", color="green")
+        plt.title("Posterior Histogram of " + str(name))
+        plt.xlabel(name)
+        plt.ylabel("Probability Density")
+        plt.legend()
+
+        if axis is not None:
+            plt.axis(axis)
+
+        if save:
+            if filename is None:
+                plt.savefig(str(name) + "_Posterior_plot.png")
+            else:
+                plt.savefig(filename)
+        return plt
+
+    def weighted_posterior_plot(
+        self, name: str, n=0, rkey=None, axis=None, save=False, filename=None
+    ):
+        """
+        Returns the weighted posterior histogram for the given parameter
```

Comment:
I'm not sure I entirely understand what the difference is between the posterior histogram and the weighted posterior histogram? Weighted by what?

Reply:
In the posterior_histogram, we use all the samples. The samples of a Nested Sampling run are not all of same weights so in the weighted_posterior histogram we use weighted samples. More info here: [Jaxns Discussion](https://github.com/Joshuaalbert/jaxns/discussions/100)

Reply:
OK, good, that makes sense!